### DISCUSSION TITLE
Tab Autocomplete not working as expected anymore

### DISCUSSION THREAD

**Discussion by artifact4727** [PowerShell/PowerShell#15612] [Enhancements]
Somewhere on the Internet I read that a .NET framework update could change the order of autocomplete.

E.g, if you have the files:
foo.bat 
foo_bar.bat
foo_bar_foo.bat

If you typed "foo "Tab key was stepping through them in descending order. Which makes sense IMHO because foo is a closer match to "foo.bat" than to "foo_bar_foo.bat". However, suddenly it started iterating backwards for me and I assume it has something to to with .NET framework as I wrote initially.  Can you still, please check if this behavior is possible to fix within Powershell Core? I would be very thankful since it makes my life easier.

**Comment by iSazonov** [comment]
Please read https://docs.microsoft.com/en-us/dotnet/standard/globalization-localization/globalization-icu

**Reply by artifact4727** [reply]
I don't seem to find anything that could help with with my question in the link? Can you give me a hint what you think I could do with the link?

**Reply by iSazonov** [reply]
The link explains the behavior you see. If you set DOTNET_SYSTEM_GLOBALIZATION_USENLS=1 you get old behavior.

**Reply by artifact4727** [reply]
Yes I tried that already. It doesn't change the completion order back. Also not after restarting Powershell.

**Reply by iSazonov** [reply]
You should set the env variable then start new pwsh process.
